src/core/model/Stroke.h

```diff
@@ -116,6 +162,12 @@ class Stroke: public AudioElement {
 protected:
     void calcSize() const override;
 
+private:
+    static std::optional<Interval<double>> intersectLineWithRectangle(const Point& p, const Point& q,
+                                                                      const xoj::util::Rectangle<double>& rectangle);
+    static TinyVector<double, 2> intersectLineSegmentWithRectangle(const Point& p, const Point& q,
```

Comment:
How does this compile? TinyVector should be incomplete.

Reply:
I think forward declaration of the return type of a **static** member function is allowed. The header for TinyVector is definitely not included, even indirectly, in this file. Its actually never included in any header file in the project.

Reply:
done